lib/Interface/c_interface.cpp

```diff
@@ -2094,3 +2094,124 @@ int vc_parseMemExpr(VC vc, const char* s, Expr* oquery, Expr* oasserts)
   }
   return 1;
 }
+
+void _vc_useSolver(VC vc, stp::UserDefinedFlags::SATSolvers solver)
+{
+  /* Helper method to encapsulate setting a solver */
+  stp::STP* stp_i = (stp::STP*)vc;
+  stp::STPMgr* b = stp_i->bm;
+  b->UserFlags.solver_to_use = solver;
+}
+
+bool _vc_isUsingSolver(VC vc, stp::UserDefinedFlags::SATSolvers solver)
+{
+  /* Helper method to encapsulate getting a solver */
+  stp::STP* stp_i = (stp::STP*)vc;
+  stp::STPMgr* b = stp_i->bm;
+  return b->UserFlags.solver_to_use == solver;
+}
+
+bool vc_supportsMinisat(__attribute__((unused)) VC vc)
+{
+  return 1;
+}
+
+bool vc_useMinisat(VC vc)
+{
+  _vc_useSolver(vc, stp::UserDefinedFlags::MINISAT_SOLVER);
+  return 1;
+}
+
+bool vc_isUsingMinisat(VC vc)
+{
+  return _vc_isUsingSolver(vc, stp::UserDefinedFlags::MINISAT_SOLVER);
+}
+
+bool vc_supportsSimplifyingMinisat(__attribute__((unused)) VC vc)
+{
+  return 1;
+}
+
+bool vc_useSimplifyingMinisat(VC vc)
+{
+  _vc_useSolver(vc, stp::UserDefinedFlags::SIMPLIFYING_MINISAT_SOLVER);
+  return 1;
+}
+
+bool vc_isUsingSimplifyingMinisat(VC vc)
+{
+  return _vc_isUsingSolver(vc, stp::UserDefinedFlags::SIMPLIFYING_MINISAT_SOLVER);
+}
+
+bool vc_supportsCryptominisat(__attribute__((unused)) VC vc)
+{
+#ifdef USE_CRYPTOMINISAT
```

Comment:
Did you catch my earlier comment about using `#if` instead of `#ifdef` here? `-DUSE_CRYPTOMINISAT=0` would do something unexpected.

Reply:
Well, there's no precent for using that in the rest of the code, but again, I can add it if it is preferable?

- https://github.com/stp/stp/blob/master/tools/stp/main.cpp#L163

- https://github.com/stp/stp/blob/master/lib/STPManager/STP.cpp#L32

(the same for Riss)

Happy to make the change, just don't know what is preferable.

Reply:
I think the other code also suffers from the same problem, but sure, we can keep it this way for consistency. That way there are no surprises where if you set it to `0`, it's not that some parts of the code will think it's available and others won't.

Reply:
That was my thoughts -- having it consistently "bad" is better (temporarily) than having it inconsistent.

At least this makes fixing the root cause (in other PR) easier.